### ISSUE TITLE
Collision polygon circle

### ISSUE THREAD

**Issue by XHXIAIEIN** [Scirra/Construct-feature-requests#187]
### Reviewed guidelines

- [X] I have read and understand the suggestion guidelines

### Checked for duplicate suggestions

- [X] I checked for existing similar suggestions

### Summary

Add a circle collision option. and adjust the radius.

### Possible workarounds or alternatives

1. add the behavior physics and put the collision in a circle.

2. Use the more points to make a collision polygon.

Why can't an octagon collision polygon replace a true circle?
Because 8 and 12 polygons may get stuck at some extreme angles when sliding.

### Proposed solution

Add a circle collision option. and adjust the radius.

### Why is this idea important?

This would solve so many problems with bullet, 8direction, car and other behaviors, that collide with solids.

Even better if we could make an ellipse collision mask with horizontal diameter=sprite.width and vertical diameter=sprite.height

**Comment by fodi** [comment]
This would be extremely useful and if I'm not mistaken checking for collisions/overlaps of objects with circle collision masks is easier/faster compared to using "circleish" polygons with a bunch of points.

I'd be very happy with the option to put even just a single circle collision mask on an object with the possibility the radius (and maybe the position) of the circle.

**Comment by Wackytoaster** [comment]
Yeah I've also heard that a check for a circle collision is really efficent. But in any case I have quite often spend time making these awkward polygon circles and it's just... bleh. Circle collision would be really elegant (box2D also has that option which is great!)
